articles/python-generators/index.md

```diff
@@ -0,0 +1,160 @@
+Generators are a special kind of function that return a lazy iterator, they were introduced with [PEP 255](https://www.python.org/dev/peps/pep-0255/). Generators are objects that you can loop through like a list.Lazy iterators do not store their contents in memory. The best way to iterate through large and complex datasets is through the use of generators.
+
+### Prerequisites
+To get started, you need to have a basic understanding of Python. All the functions and code used can be found on [repl.it](https://repl.it/@paulodhiambo962/PythonGenerators/). To execute various functions, click on **RUN**. You will now be able to access the variables and the function return values from the python interpreter. 
+
+**Generator function** is a function that returns a generator object. It is defined like a normal function, but whenever it needs to generate a value, it does so with the **yield** keyword rather than return. If the body of a def contains yield, the function automatically becomes a generator function.
+
+**Generator Object** uses lazy evaluations to yield sequences. They are used either by calling the next method on the  generator object or using the generator object in a “for in” loop.
+
+**Yield** statement suspends function’s execution and sends a value back to the caller, but retains enough state to enable function to resume  where it is left off. When resumed, the function continues execution  immediately after the last yield run. This allows its code to produce a  series of values over time, rather than computing them at once and  sending them back like a list.
+
+**Generator Expressions** are a high-performance, memory–efficient generalization of list comprehensions and generators.
+
+### Generator Expressions vs List Comprehensions
+```python
+list_comprehension = ['Hello world' for i in range(3)]
+generator_expression = ('Hello world' for i in range(3))
+```
+
+Unlike list comprehensions, generator expressions don’t  construct list objects. Instead, they generate values “just in time”  like a class-based iterator or generator function would.
```

Comment:
You will have to explain what class-based iterators or generators are. Also, can you add a section on iterators? How are they different from generators?